packages/grid/x-data-grid/src/internals/components/cell/GridEditSingleSelectCell.tsx

```diff
@@ -74,6 +75,10 @@ function GridEditSingleSelectCell(props: GridRenderEditCellParams & Omit<SelectP
     setOpen(false);
     const isValid = await api.setEditCellValue({ id, field, value: event.target.value }, event);
 
+    if (rootProps.experimentalFeatures?.enableNewEditingAPI) {
+      return;
```

Comment:
The edit cell in the current `singleSelect` column automatically commits the value after selecting an option. This doesn't work well with `preProcessEditCellProps` because if doing a AJAX validation it freeze the UI while waiting for the server to respond. Now we don't automatically commit the value. The user has to click outside as in any other column.

Reply:
If `preProcessEditCellProps` is not being used and/or the desired UX is that once a user has made a `MenuItem` selection the `singleSelect` will automatically commit, is there a prescribed method for doing so?  Or, is there a workaround to make that happen? Thank you.

Reply:
Based on the auto-stop example, my understanding is this cannot be done with the `singleSelect` column type but rather must be accomplished with a custom column type. While the custom column type may be a `Select`, it would not be a singleSelect. Thank you.